src/app/core/services/config/protocol.service.ts

```diff
@@ -24,30 +41,187 @@ export class ProtocolService {
   ) {}
 
   pull() {
-    return this.remoteConfig
-      .read()
-      .catch(e => {
-        throw this.logger.error('Failed to fetch Firebase config', e)
-      })
+    return Promise.all([this.getProjectTree(), this.getParticipantAttributes()])
+      .then(([tree, attributes]) =>
+        this.findValidProtocolUrl(tree, '', this.ATTRIBUTE_KEY, '', attributes)
+      )
```

Comment:
Can we send a firebase event on what exact protocol was chosen? On that note, maybe it will also be good to include this in the setting page (the exact protocol picked up). This will help debug any issues in production.